### PR TITLE
A dead_code! helper to find candidates for deletion

### PR DESCRIPTION
**Pull request by thermistor**
Can use `dead_code!` in controller or a view currently. 
Can include it in other code as needed.

### CODE REVIEW

File: app/controllers/admin/financials/market_payments_controller.rb [hunk 1]
```diff
@@ -3,6 +3,7 @@ class MarketPaymentsController < AdminController
     before_action :require_admin
 
     def index
+      # dead_code!
```
Comment (rbarreca) [inline app/controllers/admin/financials/market_payments_controller.rb]: Is this meant to be commented out?

File: lib/dead_code.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module DeadCode
+  NotDeadCodeError = Class.new(StandardError)
+
+  def dead_code!
+    error = NotDeadCodeError.new("NOT dead code: #{caller.first.inspect}")
```
Comment (rbarreca) [inline lib/dead_code.rb]: I'm probably being dense, but what does the exception message say "NOT dead code"?

Comment (rbarreca) [reply lib/dead_code.rb]: Nevermind, I get it now.

File: app/controllers/admin/financials/market_payments_controller.rb (hunk 1 above)
Comment (rbarreca) [reply app/controllers/admin/financials/market_payments_controller.rb]: Only real comment is this one.

Comment (thermistor) [reply app/controllers/admin/financials/market_payments_controller.rb]: I was just using this as an example of usage. I can update

**Comment by ivar** [comment]
it'd be good to add more instantiations... notably it'd be good to instrument the stripe callback stuff

**Pull request closed without merging**